src/internal/task/task_stack_cortexm.go

```diff
@@ -0,0 +1,76 @@
+// +build scheduler.tasks, cortexm
+
+package task
+
+import "unsafe"
+
+const stackSize = 1024
+
+// calleeSavedRegs is the list of registers that must be saved and restored when
+// switching between tasks. Also see scheduler_cortexm.S that relies on the
+// exact layout of this struct.
+type calleeSavedRegs struct {
+	r4  uintptr
+	r5  uintptr
+	r6  uintptr
+	r7  uintptr
+	r8  uintptr
+	r9  uintptr
+	r10 uintptr
+	r11 uintptr
+
+	pc uintptr
+}
+
+// prepareStart stores the function and arguments pointers into appropriate registers to be passed to the goroutine start function.
+func (r *calleeSavedRegs) prepareStart(fn uintptr, args unsafe.Pointer) {
+	r.pc = uintptr(unsafe.Pointer(&startTask))
+	r.r4 = fn
+	r.r5 = uintptr(args)
+}
```

Comment:
I think it makes sense to include this code directly in `archInit`, to make it easier to follow the flow of operations.